docs/design/Debug_DownloadURL.md

```diff
@@ -0,0 +1,207 @@
+
+## Non Goals
+
+- Create a CLI
+- Automatically retrieve debug information on behalf of the user
+
+## High-Level Design
+
+When a NAC velero backup or restore is completed, its associated NAB/NAR are updated to completion status.
+
+During this time, NAC will also inject into status a path (ie. S3 url) to a relevant debugging file on the storage location that velero had uploaded.
+
+## Detailed Design
+
+[Currently](https://github.com/vmware-tanzu/velero/blob/2197cab3dbf1038b7afe363284d7b9e18c9300dc/pkg/apis/velero/v1/download_request_types.go#L32-L45) velero exposes following DownloadTargetKind that can be used in a DownloadRequest spec.
+
+```go
+ DownloadTargetKindBackupLog                       DownloadTargetKind = "BackupLog"
+ DownloadTargetKindBackupContents                  DownloadTargetKind = "BackupContents"
+ DownloadTargetKindBackupVolumeSnapshots           DownloadTargetKind = "BackupVolumeSnapshots"
+ DownloadTargetKindBackupItemOperations            DownloadTargetKind = "BackupItemOperations"
+ DownloadTargetKindBackupResourceList              DownloadTargetKind = "BackupResourceList"
+ DownloadTargetKindBackupResults                   DownloadTargetKind = "BackupResults"
+ DownloadTargetKindRestoreLog                      DownloadTargetKind = "RestoreLog"
+ DownloadTargetKindRestoreResults                  DownloadTargetKind = "RestoreResults"
+ DownloadTargetKindRestoreResourceList             DownloadTargetKind = "RestoreResourceList"
+ DownloadTargetKindRestoreItemOperations           DownloadTargetKind = "RestoreItemOperations"
+ DownloadTargetKindCSIBackupVolumeSnapshots        DownloadTargetKind = "CSIBackupVolumeSnapshots"
+ DownloadTargetKindCSIBackupVolumeSnapshotContents DownloadTargetKind = "CSIBackupVolumeSnapshotContents"
+ DownloadTargetKindBackupVolumeInfos               DownloadTargetKind = "BackupVolumeInfos"
+ DownloadTargetKindRestoreVolumeInfo               DownloadTargetKind = "RestoreVolumeInfo"
+```
+
+These are the expected kinds of items we can get from object store for debugging.
+
+The path to these items in an object store is defined in [pkg/persistence/object_store_layout.go](https://github.com/vmware-tanzu/velero/blob/0a280e57863c70495a2dfb65787615a68a0e7b03/pkg/persistence/object_store_layout.go) of velero repo.
+
+We will use this knowledge to populate NAB/NAR status fields with path to log and resourceList.
+
+In the future we may have a CLI that can gather resources available in DownloadTargetKind using the values added to NAB/NAR statuses from this enhancement.
+
+### Usage examples
+
+In a completed NAB, you see in its status.resourceListPath has a value of `s3://velero-6109f5e9711c8c58131acdd2f490f451/velero-e2e-00093375-ecd8-41d8-81fc-8ce9d4983ad6/backups/mongo-blockdevice-e2e-f26ab76c-34d1-11ef-93a5-0a580a834d36/mongo-blockdevice-e2e-f26ab76c-34d1-11ef-93a5-0a580a834d36-resource-list.json.gz`
```

Comment:
Should we shorten the s3 URLs ? status would be more readable. thoughts ?

Reply:
You mean bit.ly?

Reply:
yeah something like that, but we can skip it for now, lets see if CU has any -ve feedback on this at all.